rocksdb_admin/admin_handler.cpp

```diff
@@ -815,22 +978,123 @@ void AdminHandler::async_tm_restoreDBFromS3(
   }
 
   common::Timer timer(kS3RestoreMs);
+  LOG(INFO) << "S3 Restore " << request->db_name << " from " << request->s3_backup_dir;
   auto local_s3_util = createLocalS3Util(request->limit_mbs, request->s3_bucket);
-  std::string formatted_s3_dir_path = rtrim(request->s3_backup_dir, '/');
-  rocksdb::Env* s3_env = new rocksdb::S3Env(
-  formatted_s3_dir_path, std::move(local_path), std::move(local_s3_util));
 
-  LOG(INFO) << "S3 Restore " << request->db_name << " from " << formatted_s3_dir_path;
-  if (!restoreDBHelper(request->db_name,
-                       formatted_s3_dir_path,
-                       std::unique_ptr<rocksdb::Env>(s3_env),
-                       std::move(upstream_addr),
-                       request->__isset.limit_mbs,
-                       request->limit_mbs,
-                       &e)) {
-    callback.release()->exceptionInThread(std::move(e));
-    common::Stats::get()->Incr(kS3RestoreFailure);
-    return;
+  if (FLAGS_enable_checkpoint_backup) {
```

Comment:
this is solely parallel download, not checkpoint, right?

Reply:
yes